src/main/java/org/jabref/logic/shared/DBMSConnectionProperties.java

```diff
@@ -1,20 +1,22 @@
-package org.jabref.shared;
+package org.jabref.logic.shared;
 
 import java.io.UnsupportedEncodingException;
 import java.security.GeneralSecurityException;
 import java.util.Objects;
 import java.util.Optional;
 
-import org.jabref.shared.prefs.SharedDatabasePreferences;
-import org.jabref.shared.security.Password;
+import org.jabref.logic.shared.prefs.SharedDatabasePreferences;
+import org.jabref.logic.shared.security.Password;
+import org.jabref.model.database.shared.DBMSType;
+import org.jabref.model.database.shared.DatabaseConnectionProperties;
 
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 
 /**
  * Keeps all essential data for establishing a new connection to a DBMS using {@link DBMSConnection}.
  */
-public class DBMSConnectionProperties {
+public class DBMSConnectionProperties implements DatabaseConnectionProperties {
```

Comment:
I only don't get this, isn't this supposed to be same or are DataBaseConnectionProperties more general?

Reply:
`DBMSConnectionProperties` are the only implementation of `DatabaseConnectionProperties`, they're essentially the same. The sole reason for the existence of the interface is that it breaks the dependency from model to elsewhere.